src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,197 @@
+     */
+    public void deleteTask(int index) throws DukeException {
+        if (index >= list.size()) {
+            throw new DukeException("☹ OOPS!!! I'M SORRY, CAN'T FIND TASK");
+        } else {
+            Task task = list.remove(index);
+            System.out.println("NOTED. I'VE REMOVED THIS TASK:");
+            System.out.println(task);
+            System.out.println("NOW YOU HAVE " + list.size() + " TASKS IN THE LIST.");
+        }
+    }
+
+    /**
+     * Adds to do task into our current list.
+     *
+     * @param desc the task description entered by the user
+     * @throws DukeException if task description is empty.
+     */
+    public void addToDo(String desc, boolean done) throws DukeException {
+        if (desc == null) {
+            throw new DukeException("PLEASE INCLUDE TASK DESCRIPTION IN YOUR COMMAND.");
+        }
+        Task newTask = new ToDo(desc, done);
+        list.add(newTask);
+        System.out.println("GOT IT. I'VE ADDED THIS TASK:");
+        System.out.println(newTask);
+        System.out.println("NOW YOU HAVE " + list.size() + " TASKS IN THE LIST.");
+    }
+
+    /**
+     * Adds deadline task into our current list.
+     *
+     * @param desc the task description entered by the user.
+     * @param date the date entered by the user.
+     * @param done the current status of the task
+     * @throws DukeException if task description or date is empty.
+     */
+    public void addDeadline(String desc, String date, boolean done) throws DukeException {
+        if (desc == null) {
+            throw new DukeException("PLEASE INCLUDE TASK DESCRIPTION IN YOUR COMMAND.");
+        }
+        if (date == null) {
+            throw new DukeException("PLEASE INCLUDE THE DATE IN YOUR COMMAND.");
+        }
+        try {
+            LocalDate date2 = LocalDate.parse(date);
+            Task newTask = new Deadline(desc, date2, done);
+            list.add(newTask);
+            System.out.println("GOT IT. I'VE ADDED THIS TASK:");
+            System.out.println(newTask);
+            System.out.println("NOW YOU HAVE " + list.size() + " TASKS IN THE LIST.");
+        } catch (DateTimeParseException ex) {
+            throw new DukeException("INVALID DATE FORMAT.");
+        }
+    }
+
+    /**
+     * Adds event task into our current list.
+     *
+     * @param desc the task description entered by the user.
+     * @param date the date entered by the user.
+     * @param done the current status of the task
+     * @throws DukeException if task description or date is empty.
+     */
+    public void addEvent(String desc, String date, boolean done) throws DukeException {
+        if (desc == null) {
+            throw new DukeException("PLEASE INCLUDE TASK DESCRIPTION IN YOUR COMMAND.");
+        }
+        if (date == null) {
+            throw new DukeException("PLEASE INCLUDE THE DATE IN YOUR COMMAND.");
+        }
+        try {
+            LocalDate date2 = LocalDate.parse(date);
```

Comment:
I think the variable name 'date2' is not very descriptive of what it is. At first look, I don't exactly know what date2 is. After inspecting the code, I realised that date2 is actually a parsed date. Perhaps the variable name could be called dateParsed instead? I understand variable naming is difficult when there are more important logic to handle but every small effort goes a long way for the readability of code.